card_test/venv/Scripts/code/BLOB.py

```diff
@@ -0,0 +1,119 @@
+from PIL import Image, ImageDraw
+import numpy as np
+import sys  # We need this in order to change the max recursion
+
+
+def main():
+    # Python normally only allows a function to call itself 10^4 times, which is not enough for larger BLOBs
+    sys.setrecursionlimit(10 ** 6)  # Changing the max recursion to 10^6
+    img = Image.open('Images/ace2.JPG')
+
+    img2 = binary(img)  # Converting to a binary image, based on some given benchmarks for RGB values
+    img3 = detectBlobs(img2)  # Applying the BLOB detection, which converts "burned" pixels to pink and counts big BLOBs
+
+    img3.show()
+    del img, img2, img3  # Deleting the temporary image files to save memory, since we have already shown the output
+
+
+
+def detectBlobs(img):  # This function goes through finding each BLOB and counting them
+    pixels = list(img.getdata())  # Creates a 1D list of RGB values for each pixel in the image
+    width, height = img.size  # Setting the values for width and height based on the dimensions of the image
+    pos = 0  # The position, of the current pixel, in the list
+    counter = Counter()  # Creating an object from the Counter class (which can be found further down)
+
+    for R, G, B in pixels:  # A for loop where we can work with the values of R, G & B, for the length of the pixels list
+
+        # Since we have the pixels in a 1D list, we need some values to keep track of our actual x and y positions in the image
+        # Because when "burning" pixels, we would like to track not only pixels to the left and right from our current position
+        # But also below and above, which would otherwise be hard in a 1D list
+        burnWidth = counter.widthCount  # Setting the burning position to the current x position in the image
+        burnHeight = counter.heightCount  # Setting the burning position to the current y position in the image
+
+        if (R == 255):  # Since it is a binary image, we can test if it is white using only R
+            grassFire(pos, pixels, width, height, counter, burnWidth,
+                      burnHeight)  # Starting a fire from our current position
+
+            if (counter.pixelCount > 200):  # Checking if the BLOB we just burned is big enough to be counted
+                counter.blobCount += 1  # Counting the BLOB
+            counter.pixelCount = 0  # Resetting the pixel counter, so we are ready to count the size of the next BLOB
+
+        counter.widthCount += 1  # Adding 1 to our current x position
```

Comment:
Honestly, I just want to take this opportunity to shit talk your horrible code. 💩